france_connect/clients.py

```diff
@@ -0,0 +1,192 @@
+        params = urllib.parse.urlencode(
+            {
+                "client_id": self.client_id,
+                "response_type": "code",
+                "scope": " ".join(self.scopes),
+                "acr_values": " ".join(acr_values),
+                "nonce": nonce,
+                "state": state,
+                "redirect_uri": self.post_login_callback_url,
+            }
+        )
+        url = urllib.parse.urljoin(self.fc_base_url, self.fc_authorization_url)
+        return f"{url}?{params}", nonce, state
+
+    def get_configuration(self) -> dict:
+        """Return the OpenID configuration."""
+        url = urllib.parse.urljoin(self.fc_base_url, "/api/v2/.well-known/openid-configuration")
+        try:
+            response = requests.get(
+                url, timeout=self.timeout, verify=self.verify_ssl, allow_redirects=self.allow_redirects
+            )
+            return response.json()
+        except requests.exceptions.RequestException:
+            msg = f"FranceConnect - Failed to retrieve configuration"
+            raise FranceConnectConfigurationError(response.status_code, response.text, msg)
+
+    def get_access_token(self, code: str) -> dict:
+        """Exchange the authorization code for an access token."""
+        token_url = urllib.parse.urljoin(self.fc_base_url, self.fc_token_url)
+        data = {
+            "grant_type": "authorization_code",
+            "code": code,
+            "redirect_uri": self.post_login_callback_url,
+            "client_id": self.client_id,
+            "client_secret": self.client_secret,
+        }
+        try:
+            response = requests.post(
+                token_url,
+                data=data,
+                headers={"Accept": "application/json"},
+                timeout=self.timeout,
+                verify=self.verify_ssl,
+                allow_redirects=self.allow_redirects,
+            )
+            response.raise_for_status()
+            return response.json()
+        except requests.exceptions.RequestException:
+            msg = f"FranceConnect - Failed to retrieve access_token"
+            raise FranceConnectAccessTokenError(response.status_code, response.text, msg)
+
+    def _verify_jwt(self, token: str, token_type=str) -> dict:
+        """Verify the JWT signature and returns decoded token."""
+        jwks_client = PyJWKClient(f"{self.fc_base_url}{self.fc_jwks_url}")
+        try:
+            # Get the signing key from the JWT
+            key = jwks_client.get_signing_key_from_jwt(token)
+            # Decode and verify the JWT with the signing key
+            return jwt.decode(token, key.key, algorithms=["RS256"], audience=self.client_id)
+        except ExpiredSignatureError:
+            raise FranceConnectExpiredTokenError(f"FranceConnect - FranceConnect {token_type} has expired.")
+        except InvalidTokenError:
+            raise FranceConnectInvalidTokenError(f"FranceConnect - FranceConnect invalid {token_type}")
```

Comment:
Same here, let PyJWT raise its exception since you're not adding much by catching it yourself.